src/pdf/index.ts

```diff
@@ -17,10 +17,16 @@ export type TableOfContents = {
 
 let browser: Browser;
 
+puppeteer
+  .launch({ args: ['--disable-dev-shm-usage'] })
+  .then(inst => (browser = inst))
+  .catch(e => {
+    logger.logException('Failed to start browser puppeteer', e);
+  });
+
 export async function generatePdfFromHtml(html: string) {
-  if (!browser) {
-    browser = await puppeteer.launch({ args: ['--disable-dev-shm-usage'] });
-  }
+  // TODO: create browser lazily while keeping track of it
+  // so we don't end up with dozens of browsers
 
   const name = generateTmpPath();
 
```

Comment:
Just a small comment for line 43 of this file because I see it now. Isn't it better to use `page.setContetnt()` if you want to get page from html https://github.com/puppeteer/puppeteer/blob/main/docs/api.md#pagesetcontenthtml-options. I think page.goto() is more when you want to get html from url. I don't know if I am 100% correct but as I can see on their documentation might be better choice.

Reply:
Yes, you are right, I copied that part from the blog post but `page.setContetnt()`  seems the right choice.